nginx.md

```diff
@@ -0,0 +1,190 @@
+
+Since this `location` rule uses regex, it's slightly less efficient then the rule for Pico in Document Root.  While this is unlikely to make a real-world difference, it's something to keep in mind when deciding which rule to use.
+
+#### Regex and Nginx's Processing Order
+
+When we're using a Regular Expression for a `location` block in Nginx, we need to pay close attention to the order of our config.  When analyzing your `location` blocks, Nginx first looks at any absolute locations, then tries to match against any regex rules. When it's trying to match the url to a regex rule, it stops checking after it finds the first matching rule.  The Subfolder rule above is very *very* broad.  It matches *anything* inside the `pico` subfolder.  Because of this, it also matches any `.php` files, `.htaccess`, and any of the folders we denied access to earlier.
+
+This means that the Pico rewrite rule must come **last** in your server configuration.  If any of the other rules we've defined are after it, they will never be used.  Nginx will stop once it gets to the rewrite rule since it will be the first match.
+
+## Example Configuration
+
+If we combine all the examples above, your configuration will look something like this:
+
+```
+server {
+	listen 80;
+
+	server_name www.example.com example.com;
+	root /var/www/example;
+
+	index index.php;
+
+	location ~ /(\.htaccess|\.git|config|content|content-sample|lib|vendor) {
+		return 404;
+	}
+
+	location ~ [^/]\.php(/|$) {
+		fastcgi_split_path_info ^(.+?\.php)(/.*)$;
+
+		# Protection Against "cgi.fix_pathinfo = 1"
+		if (!-f $document_root$fastcgi_script_name) {
+			return 404;
+		}
+
+		fastcgi_pass unix:/var/run/php5-fpm.sock;
+		fastcgi_index index.php;
+		include fastcgi_params;
+
+		# Inform Pico we will be rewriting URL's for it.
+		fastcgi_param PICO_URL_REWRITING 1;
+	}
+
+	location / {
+		try_files $uri $uri/ /?$uri&$args;
+	}
+}
+```
+
+Again, please note that this is only provided as an **example**.  You should not copy it, but only use it as a reference.  Your own configuration will depend very much on your own system configuration.  If you do copy this code, be sure to modify it according to your own requirements.
+
+## Advanced Configuration Tips
+
+### Modular Pico Config
+
+Let's say you're a real Pico enthusiast and have several Pico websites running on the same server.  You may get tired of writing all these rules into each and every server configuration.  An easier solution might be to place all the common components (index, access denials, php rules) into a separate file and include it using `include /absolute/path/to/file`.  You could also add the rewrite rule to this file, but a better option would be to include a second file, that way you can chose to include it *only* when Pico is located in your Document Root.
```

Comment:
I think it's a good idea to let the components, i.e.

> An easier solution might be to place all the common components (index, access denials, php rules) into a separate file

directly link to the appropriate sections above :smiley: